overlord/hookstate/hookmgr.go

```diff
@@ -256,24 +267,30 @@ func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	if hookExists {
-		output, err := runHook(context, tomb)
-		if err != nil {
-			if hooksup.TrackError {
-				trackHookError(context, output, err)
-			}
-			err = osutil.OutputErr(output, err)
-			if hooksup.IgnoreError {
-				task.State().Lock()
-				task.Errorf("ignoring failure in hook %q: %v", hooksup.Hook, err)
-				task.State().Unlock()
-			} else {
-				if handlerErr := context.Handler().Error(err); handlerErr != nil {
-					return handlerErr
-				}
-
-				return fmt.Errorf("run hook %q: %v", hooksup.Hook, err)
+	// the core snap is special and is handled differently
+	var output []byte
+	if hooksup.Snap == "core" && hooksup.Hook == "configure" {
```

Comment:
we probably want a TODO here about having a general RegisterHijack mechanism or something later

Reply:
I pused a version that uses the hijacker now.